drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/DMNTypeServiceImpl.java

```diff
@@ -122,14 +124,18 @@ protected FactModelTree createTopLevelFactModelTree(String factName, DMNType typ
      * @return
      * @throws WrongDMNTypeException if <code>DMNType.isCollection()</code> != <code>true</code>
      */
-    protected FactModelTree createFactModelTreeForCollection(Map<String, List<String>> genericTypeInfoMap, String factName, DMNType type, SortedMap<String, FactModelTree> hiddenFacts, FactModelTree.Type fmType) throws WrongDMNTypeException {
-        if (!type.isCollection() && !isToBeManagedAsCollection((BaseDMNTypeImpl) type)) {
+    protected FactModelTree createFactModelTreeForCollection(Map<String, List<String>> genericTypeInfoMap, String factName, DMNType type, SortedMap<String, FactModelTree> hiddenFacts, FactModelTree.Type fmType, List<String> alreadyVisited) throws WrongDMNTypeException {
+        if (!type.isCollection() && !isToBeManagedAsCollection(type)) {
             throw new WrongDMNTypeException();
         }
-        populateGeneric(genericTypeInfoMap, "value", type.getName());
+        String typeName = type.getName();
+        populateGeneric(genericTypeInfoMap, "value", typeName);
```

Comment:
@danielezonca Please put this "value" string in a static final constant, considering it's used in multiple points.

Reply:
Done, refactored and created this PR https://github.com/kiegroup/drools/pull/2594